ultravox/training/configs/stage2_lora.yaml

```diff
@@ -1,16 +1,23 @@
+exp_name: stage2_lora__gs_ai_bq
+
 text_model_lora_config:
   r: 64  # no/little change in the range [16, 64]
   target_modules: ['mlp.gate_proj', 'mlp.up_proj', 'mlp.down_proj', 'v_proj', 'o_proj', 'k_proj', 'q_proj']
 
-data_sets: ["commonvoice", "peoplespeech", "anyinstruct"]
+data_sets: ["gigaspeech", "anyinstruct", "boolq_extended"]
 
 # disable_layer_drop: True
 # audio_model_lora_config:
 #   r: 64
 #   target_modules: ['k_proj', 'q_proj', 'v_proj', 'out_proj', 'intermediate_dense', 'output_dense']
 
-num_prompts: 6
+num_prompts: 11
 
 lr: 1.e-4  # need a lower LR for LLM fine-tuning
+lr_scheduler: constant_with_warmup
```

Comment:
why not cosine?

Reply:
This is a very old change tbh, I just never pushed it.

Two reasons:
  1) with constant the val_loss metrics actually went lower
  2) I've always wondered if our models are under-trained, but with cosine, looking at the loss I kept thinking that training had almost converged, when in fact it had not.

Even if evidence for (1) wasn't there (and if in the future it reverses), I think (2) alone can cause issues as we are still exploring the subject. Maybe our final (stable) training runs should use cosine (if it turns out to be better), but not our exploratory runs.

Graphs for cosine (purple) vs constant (yellow):
![image](https://github.com/fixie-ai/ultravox/assets/5062458/c98f2106-b2c4-4813-8ba6-601d705578f8)

You can see eval/loss going way lower with yellow, and train/loss is actually going up for some reason.

I then let the training run for much longer and the training did not converge even with 8x the amount of training.
I should note that the test metrics didn't support that training longer leads to much benefits so I haven't touched `max_steps` for our exploratory train jobs. The v0.2 model was trained for longer though.